python/tests/io/test_pyarrow.py

```diff
@@ -1014,7 +1014,7 @@ def test_projection_nested_new_field(file_struct: str) -> None:
     result_table = project(schema, [file_struct])
 
     for actual, expected in zip(result_table.columns[0], [None, None, None]):
-        assert actual.as_py() == {"null": expected}
+        assert actual.as_py() == expected
```

Comment:
I think this change is incorrect. There is a non-null location in every row so there should be a non-null struct for each result. The null/None value is the "null" column that doesn't exist.